public/javascripts/ood_shell.js

```diff
@@ -55,53 +8,52 @@ function OodShell(element, url, prefs) {
   this.term    = null;
 }
 
+/**
+ * Create the terminal. The constructor for the OodShell class is mostly just a data object. This function
+ * does the actual work of initializing the hterm.Terminal, the websocket (through a callback *after* the 
+ * terminal is ready.) and setting preferences, etc.
+ */
 OodShell.prototype.createTerminal = function () {
-  if(lib.resource.get('hterm/changelog/version', null).data === '1.80') {
-    this.socket = new WebSocket(this.url);
-    this.socket.onopen    = this.runTerminal.bind(this);
-    this.socket.onmessage = this.getMessage.bind(this);
-    this.socket.onclose   = this.closeTerminal.bind(this);
-  } else {
-    document.getElementById('terminal').innerHTML = 'WARNING: version of hterm has changed. Please review ood_shell.js to ensure that the override of hterm.Terminal.prototype.copyStringToClipboard is still an appropriate fix to copy on select in Firefox.';
-  }
-};
-
-OodShell.prototype.runTerminal = function () {
-  var that = this;
-
   // Set backing store that hterm uses to read/write preferences
   hterm.defaultStorage = new lib.Storage.Memory();
-
+  
   // Create an instance of hterm.Terminal
   this.term = new hterm.Terminal();
-
-  // Set preferences for terminal
-  for (var k in this.prefs) {
-    this.term.prefs_.set(k, this.prefs[k]);
-  }
-
+    
+  var that = this;
+    
   // Handler that fires when terminal is initialized and ready for use
   this.term.onTerminalReady = function () {
     // Create a new terminal IO object and give it the foreground.
     // (The default IO object just prints warning messages about unhandled
     // things to the JS console.)
     var io = this.io.push();
-
+  
     // Set up event handlers for io
     io.onVTKeystroke    = that.onVTKeystroke.bind(that);
     io.sendString       = that.sendString.bind(that);
     io.onTerminalResize = that.onTerminalResize.bind(that);
-
+    
+    that.socket = new WebSocket(that.url);  // WS isn't made or bound until terminal is ready
+    //that.socket.onopen    = nothing left to do, no onopen callback bc the WS is the last thing to init
+    that.socket.onmessage = that.getMessage.bind(that);
+    that.socket.onclose   = that.closeTerminal.bind(that);
+    
```

Comment:
Websockets are now being initialized only when the terminal is ready. After the 1.85 upgrade Firefox had some issues with all the callbacks going on. So, in an effort to reduce the shear amount of callbacks I left it to initialize the terminal first (which is asynchronous) *then* when it's ready initialize the websocket. Leaving a note there that there's no need for the socket.onopen callback because it's the last thing we need created.